ui/App/ProfileScreen.svelte

```diff
@@ -6,50 +6,165 @@
+  const session = Session.unsealed();
 
-  import CreateProjectModal from "ui/App/CreateProjectModal.svelte";
+  interface ProfileProjects {
+    cloned: project.Project[];
+    follows: project.Project[];
+    requests: project.Request[];
+  }
 
-  export let activeTab: router.ProfileTab;
-
-  const tabs = (active: router.ProfileTab) => {
-    return [
-      {
-        title: "Projects",
-        active: active === "projects",
-        icon: Icon.ChevronLeftRight,
-        onClick: () => {
-          router.push({ type: "profile", activeTab: "projects" });
-        },
-      },
-      {
-        title: "Following",
-        active: active === "following",
-        icon: Icon.Network,
-        onClick: () => {
-          router.push({ type: "profile", activeTab: "following" });
-        },
-      },
-    ];
+  const profileProjectsStore = remote.createStore<ProfileProjects>();
+  profileProjectsStore.loading();
+
+  const fetchProfileProjectsExecutor = mutexExecutor.create();
+
+  fetchProfileProjects();
+  const unsubPeerEvents = localPeer.requestEvents.subscribe(() => {
+    fetchProfileProjects();
+  });
+  onDestroy(unsubPeerEvents);
+
+  async function fetchProfileProjects(): Promise<void> {
+    try {
+      const profileProjects = await fetchProfileProjectsExecutor.run(
+        async () => {
+          const [cloned, follows, allRequests] = await Promise.all([
+            proxy.client.project.listContributed(),
+            proxy.client.project.listTracked(),
+            proxy.client.project.requestsList(),
+          ]);
+          const requests = allRequests.filter(
+            req =>
+              req.type !== project.RequestStatus.Cloned &&
+              req.type !== project.RequestStatus.Cancelled &&
+              req.type !== project.RequestStatus.TimedOut
+          );
+          return { cloned, follows, requests };
+        }
+      );
+      if (profileProjects) {
+        profileProjectsStore.success(profileProjects);
+      }
+    } catch (err: unknown) {
+      error.show(
+        new error.Error({ message: "Failed to fetch projects.", source: err })
+      );
+    }
+  }
+
+  const showNotificationsForFailedProjects = async (): Promise<void> => {
+    try {
+      const failedProjects = await proxy.client.project.listFailed();
+      failedProjects.forEach(failedProject => {
+        error.show(
+          new error.Error({
+            code: error.Code.ProjectRequestFailure,
+            message: `The project ${failedProject.metadata.name} couldn’t be loaded`,
+            details: failedProject,
+          })
+        );
+      });
+    } catch (err: unknown) {
+      error.show(
+        new error.Error({
+          code: error.Code.ProjectRequestFailure,
+          message: "Failed to get failed projects",
+          source: err,
+        })
+      );
+    }
   };
 
-  const session = Session.unsealed();
+  const onSelect = ({ detail: project }: { detail: Project }) => {
```

Comment:
I’d prefer if the function name describes what the function does instead of where it is used. We also prefer `function` statements to give us more flexibility when ordering the functions.

```suggestion
  function openProject ({ detail: project }: { detail: Project }) {
```

Reply:
https://github.com/radicle-dev/radicle-upstream/pull/2398/commits/485ac38bc0f582ae01cf121375c22e1245297692